internal/tlscert/tls_cert_validator.go

```diff
@@ -142,6 +173,18 @@ func parsePrivateKey(keyPEM []byte) error {
 	return nil
 }
 
+func validateCA(cert *x509.Certificate, ca *x509.Certificate) error {
```

Comment:
What happens if the CA bundle contains a chain of root CA and intermediate CA? The check for a matching issuer below might fail in this case as well.

Reply:
I will remove the `Issuer == Subject` check